pytorch_lightning/trainer/trainer.py

```diff
@@ -2473,6 +2474,33 @@ def _active_loop(self) -> Optional[Union[FitLoop, EvaluationLoop, PredictionLoop
     Logging properties
     """
 
+    @property
+    def logger(self) -> Optional[LightningLoggerBase]:
+        return self._logger
+
+    @logger.setter
+    def logger(self, new_logger: Optional[LightningLoggerBase]) -> None:
+        self.loggers = [new_logger] if new_logger else None
+
+    @property
+    def loggers(self) -> List[LightningLoggerBase]:
+        return self._loggers
+
+    @loggers.setter
+    def loggers(self, new_loggers: Optional[Iterable[LightningLoggerBase]]) -> None:
+        if new_loggers:
+            self._loggers = list(new_loggers)
+            if len(self._loggers) == 1:
+                new_logger = self._loggers[0]
+                if isinstance(new_logger, LoggerCollection):
```

Comment:
I don't think we need to handle the case where someone does:
`trainer.loggers = [LoggerCollection()]`

Reply:
This is actually to handle the special case where someone says `trainer.logger = LoggerCollection([logger])` (aka a LoggerCollection of size 1). The `trainer.logger` setter will pass [LoggerCollection()] to the `trainer.loggers` setter. There are a bunch of ways to handle this special case but they are all pretty messy and this is the cleanest one I could think of.

Reply:
Ok, I see. Tbh I prefer how it was originally, although it had more LOC it was more readable and easier to understand. I may be in the minority though, and if so we can move forward with this.

Reply:
No, I agree with you on this one. In this case, while having one setter call other is very elegant and reduces LOC, it obfuscates the important logic which makes it harder to follow. Since we are deprecating trainer.logger in favor of trainer.loggers, I think it's important for this part to be readable for users trying to adapt their code. Tagging @carmocca so you know why I'm changing it back.

Reply:
One pro of one setter calling the other is that it reduces the surface for bugs. Especially considering that the callee is the setter that will stay in the long term.

But I'm fine either way. You can keep it like this if you both prefer the current format

Reply:
Thank you @carmocca! I will keep this in mind for the future :)
I think for this case calling one setter in another is hard to do cleanly because their inputs are sometimes the same and sometimes different (specifically LoggerCollection, which is both `Iterable[LightningLoggerBase]` and `LightningLoggerBase`). There are different behaviors depending on which setter was called, but there is no simple way to transfer that information from one to another.